### PR TITLE
fix: enable foreign keys for sqlite

### PR DESCRIPTION
**Pull request by sylvlecl**
By default, sqlite does not enforce foreign key constraints, this needs to be enabled explicitly.

This is in particular important to catch those errors in our unit tests, but also for 
desktop mode.

**Note**
I have checked that indeed, tests before [this fix](https://github.com/AntaresSimulatorTeam/AntaREST/pull/2169) did not pass.

### CODE REVIEW

File: antarest/core/utils/fastapi_sqlalchemy/middleware.py [hunk 1]
```diff
@@ -16,6 +16,24 @@
 _session: ContextVar[Optional[Session]] = ContextVar("_session", default=None)
 
 
+def _is_sqlite_connection(dbapi_connection: Any) -> bool:
+    cls = dbapi_connection.__class__
+    full_classe_name = f"{cls.__module__}{cls.__name__}"
+    return "sqlite" in full_classe_name.lower()
```
Comment (MartinBelthle) [inline antarest/core/utils/fastapi_sqlalchemy/middleware.py]: I imagine there's not a way to do this in a cleaner way (check of class instance) :/ ?

**Review by MartinBelthle — DISMISSED** [review]

File: antarest/core/utils/fastapi_sqlalchemy/middleware.py (hunk 1 above)
Comment (sylvlecl) [reply antarest/core/utils/fastapi_sqlalchemy/middleware.py]: Check of class instance will be more fragile, this is kind of an implementation detail of sqlalchemy driver, I prefer being robust to any implementation change.
It's unlikely that another driver will have sqlite in its name.

Comment (MartinBelthle) [reply antarest/core/utils/fastapi_sqlalchemy/middleware.py]: Ok I see

Comment (sylvlecl) [reply antarest/core/utils/fastapi_sqlalchemy/middleware.py]: I changed the logic to depend on the driver name instead, it should be a little  more reliable than class names.

**Pull request merged**